scipy/optimize/tests/test_linprog.py

```diff
@@ -359,6 +358,48 @@ def f(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None):
 
         assert_raises(ValueError, f, [1, 2], A_ub=np.zeros((1, 1, 3)), b_eq=1)
 
+    def test_bounds_fixed(self):
+
+        def g(c, bounds):
+            res = linprog(c, bounds=bounds, method=self.method, options=self.options)
+            return res
+
+        # Test fixed bounds (upper equal to lower)
+        res = g([1], bounds=(1, 1))
+        _assert_success(res, 1, 1)
+        res = g([1, 2, 3], bounds=[(5, 5), (-1, -1), (3, 3)])
+        _assert_success(res, 12, [5, -1, 3])
+        res = g([1, 1], bounds=[(1, 1), (1, 3)])
+        _assert_success(res, 2, [1, 1])
+
+    def test_bounds_infeasible(self):
+
+        def g(c, bounds):
+            res = linprog(c, bounds=bounds, method=self.method, options=self.options)
+            return res
+
+        # Test ill-valued bounds (upper less than lower, lower inf, upper -inf)
+        res = g([1], bounds=(1, -2))
+        _assert_infeasible(res)
+        res = g([1], bounds=[(1, -2)])
+        _assert_infeasible(res)
+        res = g([1, 2, 3], bounds=[(5, 0), (1, 2), (3, 4)])
+        _assert_infeasible(res)
+
+        dopr = self.options.get('presolve', True)
+        if not dopr and self.method == 'simplex':
+            # For the simplex method, the cases below do not result in an
+            # infeasible status, but in a RuntimeWarning. This is a
+            # consequence of having _presolve() take care of feasibility
+            # checks.
```

Comment:
We're discussing this in gh-11618; I hope it is going to be fixed.

Reply:
That would simplify this test. I'll refer to this issue in the test.